wire-ios-data-model/Tests/Source/EAR/EARServiceTests.swift

```diff
@@ -0,0 +1,645 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import XCTest
+import LocalAuthentication
+@testable import WireDataModel
+
+final class EARServiceTests: ZMBaseManagedObjectTest, EARServiceDelegate {
+
+    var sut: EARService!
+    var keyRepository: MockEARKeyRepositoryInterface!
+    var keyEncryptor: MockEARKeyEncryptorInterface!
+
+    // MARK: - Life cycle
+
+    override func setUp() {
+        super.setUp()
+
+        keyRepository = MockEARKeyRepositoryInterface()
+        keyEncryptor = MockEARKeyEncryptorInterface()
+
+        sut = EARService(
+            accountID: userIdentifier,
+            keyRepository: keyRepository,
+            keyEncryptor: keyEncryptor,
+            databaseContexts: [uiMOC, syncMOC]
+        )
+
+        sut.delegate = self
+        prepareForMigrationCalls = 0
+
+        createSelfClient(onMOC: uiMOC)
+    }
+
+    override func tearDown() {
+        sut = nil
+        keyRepository = nil
+        keyEncryptor = nil
+        super.tearDown()
+    }
+
+    // MARK: - Delegate
+
+    var prepareForMigrationCalls = 0
```

Comment:
move to top of class

Reply:
done in 6be38e6a0a105db78177426987ca9a5366c4eeab